query.go

```diff
@@ -90,22 +91,44 @@ func getErrorFrameFromQuery(query *Query) data.Frames {
 	return frames
 }
 
+func queryContext(ctx context.Context, db *sql.DB, converters []sqlutil.Converter, fillMode *data.FillMissing, q *Query) (data.Frames, error) {
+	var (
+		done = make(chan bool)
+		res  = data.Frames{}
+		err  error
+	)
+
+	go func() {
+		res, err = query(ctx, db, converters, fillMode, q)
+		done <- true
+	}()
+
+	select {
+	case <-done:
+		// The done channel will receive data if the query funciton has completed
+		return res, err
+	case <-ctx.Done():
+		// The context's done channel will receive data if the timeout is exceeded
+		return nil, ErrorTimeout
+	}
+}
```

Comment:
isn't this `ctx` handling given for free just using `db.QueryContext`? (I may be looking to the wrong place but it seems handled the same way [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/database/sql/ctxutil.go;l=57;drc=refs%2Ftags%2Fgo1.17))

Reply:
You may be right... if the context is cancelled then the `QueryContext` function may just return an error. I'll look into this.